Issue from rock5/rock5c/app-development/moonlight

URL: [https://docs.radxa.com/rock5/rock5c/app-development/moonlight](https://docs.radxa.com/rock5/rock5c/app-development/moonlight)

Time: 29/01/2026, 17:28:28


The steps result on moonlight compilation linker error:

```
make[2]: Entering directory '/home/radxa/moonlight-qt/app'
g++ -Wl,-rpath-link,/usr/lib/aarch64-linux-gnu -o moonlight  release/nvaddress.o release/nvapp.o release/pair.o release/main.o release/computerseeker.o release/identitymanager.o release/nvcomputer.o release/nvhttp.o release/nvpairingmanager.o release/computermanager.o release/boxartmanager.o release/richpresencemanager.o release/commandlineparser.o release/listapps.o release/quitstream.o release/startstream.o release/compatfetcher.o release/mappingfetcher.o release/streamingpreferences.o release/abstouch.o release/gamepad.o release/input.o release/keyboard.o release/mouse.o release/reltouch.o release/session.o release/audio.o release/sdlaud.o release/computermodel.o release/appmodel.o release/bandwidth.o release/streamutils.o release/autoupdatechecker.o release/path.o release/mappingmanager.o release/sdlgamepadkeynavigation.o release/overlaymanager.o release/systemproperties.o release/wm.o release/ffmpeg.o release/genhwaccel.o release/sdlvid.o release/swframemapper.o release/pacer.o release/vaapi.o release/vdpau.o release/drm.o release/masterhook.o release/masterhook_internal.o release/eglvid.o release/egl_extensions.o release/eglimagefactory.o release/waylandvsyncsource.o release/qrc_resources.o release/qrc_qml.o release/moc_pair.o release/moc_compatfetcher.o release/moc_mappingfetcher.o release/moc_computerseeker.o release/moc_nvhttp.o release/moc_computermanager.o release/moc_boxartmanager.o release/moc_listapps.o release/moc_quitstream.o release/moc_startstream.o release/moc_streamingpreferences.o release/moc_session.o release/moc_computermodel.o release/moc_appmodel.o release/moc_autoupdatechecker.o release/moc_sdlgamepadkeynavigation.o release/moc_systemproperties.o   -ldl -pthread -L/home/radxa/moonlight-qt/app/../moonlight-common-c/ -lmoonlight-common-c -L/home/radxa/moonlight-qt/app/../qmdnsengine/ -lqmdnsengine -L/home/radxa/moonlight-qt/app/../h264bitstream/ -lh264bitstream -L/usr/lib -lavformat -lssl -lcrypto -lSDL2_ttf -lSDL2 -lopus -lavcodec -lavutil -lswscale -ldrm -lX11 -lva-x11 -lva-wayland -lwayland-client -lva-drm -lva /usr/lib/aarch64-linux-gnu/libQt6QuickControls2.so /usr/lib/aarch64-linux-gnu/libQt6Quick.so /usr/lib/aarch64-linux-gnu/libQt6OpenGL.so /usr/lib/aarch64-linux-gnu/libQt6Svg.so /usr/lib/aarch64-linux-gnu/libQt6Gui.so /usr/lib/aarch64-linux-gnu/libGLX.so /usr/lib/aarch64-linux-gnu/libOpenGL.so /usr/lib/aarch64-linux-gnu/libQt6QmlModels.so /usr/lib/aarch64-linux-gnu/libQt6Qml.so /usr/lib/aarch64-linux-gnu/libQt6Network.so /usr/lib/aarch64-linux-gnu/libQt6Core.so -lpthread -lGLX -lOpenGL -lEGL
/usr/bin/ld: /usr/lib/libavcodec.a(tiff.o): undefined reference to symbol 'inflateEnd'
/usr/bin/ld: /usr/lib/aarch64-linux-gnu/libz.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

--- a/docs/common/dev/_moonlight.mdx
+++ b/docs/common/dev/_moonlight.mdx
@@ -12,6 +12,12 @@
 参考至[编译并安装 ffmpeg-rockchip](rtsp?target=ffmpeg#编译并安装-ffmpeg-rockchip)
 即可安装 rockchip-ffmpeg。
 
+:::tip
+如果你是自行编译 `ffmpeg-rockchip` 给 Moonlight 使用，请确保 `ffmpeg-rockchip` 的 `./configure`
+参数包含 `--enable-shared`。否则 Moonlight 在链接阶段可能会误用静态 `libavcodec.a`，并报出类似
+`undefined reference to symbol 'inflateEnd'` 这样的缺少 zlib 依赖错误。
+:::
+
 ## 安装 Moonlight 编译依赖
 
 ```bash

--- a/i18n/en/docusaurus-plugin-content-docs/current/common/dev/_moonlight.mdx
+++ b/i18n/en/docusaurus-plugin-content-docs/current/common/dev/_moonlight.mdx
@@ -12,6 +12,12 @@ but the results may be unsatisfactory.
 Refer to [RTSP Streaming](rtsp?target=ffmpeg), there is no need to follow the entire documentation. Just follow the instructions up to
 [Compile and Install ffmpeg-rockchip](rtsp?target=ffmpeg#compile-and-install-ffmpeg-rockchip) to install rockchip-ffmpeg.
 
+:::tip
+If you build `ffmpeg-rockchip` yourself for Moonlight, make sure the `ffmpeg-rockchip` `./configure`
+arguments include `--enable-shared`. Otherwise Moonlight may pick the static `libavcodec.a` during
+linking and fail with missing system library errors such as `undefined reference to symbol 'inflateEnd'`.
+:::
+
 ## Install Moonlight build dependencies
 
 ```bash

